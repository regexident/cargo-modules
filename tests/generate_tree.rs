#[macro_use]
mod util;

mod smoke {
    test_cmd!(
        args: "generate tree",
        success: true,
        color_modes: ColorModes::ALL,
        project: smoke
    );
}

mod default {
    mod pass {
        test_cmds!(
            args: "generate tree",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                package_lib_target,
                virtual_workspace_single_package_lib_target,
                virtual_workspace_single_package_bin_target,
                workspace_single_package_lib_target,
                workspace_single_package_bin_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree",
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_multi_target,
                virtual_workspace_single_package_multi_target,
                workspace_single_package_multi_target,
            ]
        );
    }
}

mod lib {
    mod pass {
        test_cmds!(
            args: "generate tree \
                    --lib",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_lib_target,
                package_multi_target,
                virtual_workspace_single_package_lib_target,
                virtual_workspace_single_package_multi_target,
                workspace_single_package_lib_target,
                workspace_single_package_multi_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree \
                    --lib", // does not exist
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                virtual_workspace_single_package_bin_target,
                workspace_single_package_bin_target,
            ]
        );
    }
}

mod bin {
    mod pass {
        test_cmds!(
            args: "generate tree \
                    --bin package_bin_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                virtual_workspace_single_package_bin_target,
                workspace_single_package_bin_target,
            ]
        );

        test_cmds!(
            args: "generate tree \
                    --bin package_multi_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_multi_target,
                virtual_workspace_single_package_multi_target,
                workspace_single_package_multi_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree \
                    --bin foobar", // does not exist
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_lib_target,
                package_multi_target,
                virtual_workspace_multi_package,
                virtual_workspace_single_package_lib_target,
                virtual_workspace_single_package_multi_target,
                workspace_multi_package,
                workspace_single_package_lib_target,
                workspace_single_package_multi_target,
            ]
        );
    }
}

mod package {
    mod pass {
        test_cmds!(
            args: "generate tree \
                    --package package_lib_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_lib_target,
                virtual_workspace_single_package_lib_target,
                workspace_single_package_lib_target,
            ]
        );

        test_cmds!(
            args: "generate tree \
                    --package package_bin_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                virtual_workspace_single_package_bin_target,
                workspace_single_package_bin_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree \
                    --package foobar",
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                package_lib_target,
                package_multi_target,
                virtual_workspace_multi_package,
                virtual_workspace_single_package_bin_target,
                virtual_workspace_single_package_lib_target,
                virtual_workspace_single_package_multi_target,
                workspace_multi_package,
                workspace_single_package_bin_target,
                workspace_single_package_lib_target,
                workspace_single_package_multi_target,
            ]
        );
    }
}

mod package_lib {
    mod pass {
        test_cmds!(
            args: "generate tree \
                    --package package_lib_target \
                    --lib",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_lib_target,
                workspace_single_package_lib_target,
                virtual_workspace_single_package_lib_target,
            ]
        );

        test_cmds!(
            args: "generate tree \
                    --package package_multi_target \
                    --lib",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_multi_target,
                workspace_single_package_multi_target,
                virtual_workspace_single_package_multi_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree \
                    --package package_bin_target \
                    --lib", // does not exist
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                workspace_single_package_bin_target,
                virtual_workspace_single_package_bin_target,
            ]
        );
    }
}

mod package_bin {
    mod pass {
        test_cmds!(
            args: "generate tree \
                    --package package_bin_target \
                    --bin package_bin_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                workspace_single_package_bin_target,
                virtual_workspace_single_package_bin_target,
            ]
        );

        test_cmds!(
            args: "generate tree \
                    --package package_multi_target \
                    --bin package_multi_target",
            success: true,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_multi_target,
                workspace_single_package_multi_target,
                virtual_workspace_single_package_multi_target,
            ]
        );
    }

    mod fail {
        test_cmds!(
            args: "generate tree \
                    --package workspace-package \
                    --bin foobar", // does not exist
            success: false,
            color_modes: ColorModes::PLAIN,
            projects: [
                package_bin_target,
                package_lib_target,
                package_multi_target,
                virtual_workspace_multi_package,
                virtual_workspace_single_package_bin_target,
                virtual_workspace_single_package_lib_target,
                virtual_workspace_single_package_multi_target,
                workspace_multi_package,
                workspace_single_package_bin_target,
                workspace_single_package_lib_target,
                workspace_single_package_multi_target,
            ]
        );
    }
}

mod with_orphans {
    test_cmd!(
        args: "generate tree \
                --with-orphans",
        success: true,
        color_modes: ColorModes::PLAIN,
        project: smoke
    );
}

mod with_tests {
    test_cmd!(
        args: "generate tree \
                --with-tests",
        success: true,
        color_modes: ColorModes::PLAIN,
        project: smoke
    );
}

mod with_types {
    test_cmd!(
        args: "generate tree \
                --with-types",
        success: true,
        color_modes: ColorModes::PLAIN,
        project: smoke
    );
}
