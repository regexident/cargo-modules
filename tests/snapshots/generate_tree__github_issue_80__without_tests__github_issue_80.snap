---
source: tests/generate_tree.rs
expression: output

---
STDERR:

STDOUT:

crate github_issue_80
├── struct OnlyExistsWithoutTest: pub #[cfg(not(test))]
├── struct Placebo: pub
├── mod imported: pub
│   ├── struct OnlyUsedWithTest: pub
│   ├── struct OnlyUsedWithoutTest: pub
│   └── struct Placebo: pub
├── mod importing: pub
└── mod only_exists_without_test: pub #[cfg(not(test))]
    ├── struct OnlyExistsWithoutTest: pub #[cfg(not(test))]
    └── struct Placebo: pub

